src/os/bluestore/BlueStore.cc

```diff
@@ -1391,6 +1391,10 @@ void BlueStore::BufferSpace::finish_write(Cache* cache, uint64_t seq)
 {
   std::lock_guard<std::recursive_mutex> l(cache->lock);
 
+  if (writing.empty() || writing.rbegin()->seq > seq) {
```

Comment:
the writing.rbegin()->seq > seq part looks wrong.. there would still be an item == seq in writing.